<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Brian J. Knaus" />


<title>Methods</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Vining Lab</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="black_raspberry.html">Black Raspberry</a>
</li>
<li>
  <a href="hazelnut.html">Hazelnut</a>
</li>
<li>
  <a href="hemp.html">Hemp</a>
</li>
<li>
  <a href="mint.html">Mint</a>
</li>
<li>
  <a href="potato.html">Potato</a>
</li>
<li>
  <a href="methods.html">Methods</a>
</li>
<li>
  <a href="reading_group.html">Reading group</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Methods</h1>
<h4 class="author">Brian J. Knaus</h4>
<h4 class="date">7/28/2021</h4>

</div>


<div id="blast-notes" class="section level2">
<h2>Blast notes</h2>
<p>For simple tasks we may just go to NCBI: <a
href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">NCBI BLAST</a>.</p>
</div>
<div id="links" class="section level2">
<h2>Links</h2>
<p><a
href="https://wheat.pw.usda.gov/demos/BatchPrimer3/">BatchPrimer3</a><br />
<a href="https://cran.r-project.org/">CRAN</a><br />
<a href="http://dia-installer.de/">Dia</a> diagram editor<br />
<a href="https://www.gimp.org/">GIMP</a><br />
<a href="https://docs.github.com/en">GitHub docs</a><br />
<a
href="https://resources.github.com/whitepapers/github-and-rstudio/">GitHub
Pages</a><br />
<a
href="https://excellenceinbreeding.org/toolbox/tools/helium">Helium</a>
Pedigree visualization<br />
<a href="https://inkscape.org/">Inkscape</a><br />
<a href="https://www.libreoffice.org/">LibreOffice</a><br />
<a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">NCBI BLAST</a>.<br />
<a href="(https://phytozome-next.jgi.doe.gov)">Phytozome</a><br />
<a href="https://www.rstudio.com/resources/cheatsheets/">RStudio
Cheatsheets</a><br />
<a href="https://www.scribus.net/">Scribus</a> desktop publishing</p>
</div>
<div id="gtf-files" class="section level2">
<h2>GTF files</h2>
<p>A common format for storing biological coordinate data, such as the
chromosome, start and end position of a feature, is the GTF/GFF format.
The General Feature Format (GFF) and the General Transfer Format (GTF)
are similar formats that have technical differences. These technical
details should be found online, such as <a
href="https://en.wikipedia.org/wiki/General_feature_format">GFF on
Wikipedia</a> and <a
href="https://useast.ensembl.org/info/website/upload/gff.html">GFF/GTF
File Format at ensembl</a>. <strong>Table 1</strong> is an example GTF
file produced by the <a
href="https://github.com/Gaius-Augustus/BRAKER">BRAKER pipeline</a>.</p>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:600px; overflow-x: scroll; width:800px; ">
<table class="table table-striped" style="margin-left: auto; margin-right: auto;">
<caption>
<strong>Table 1.</strong> Example GTF file produced by BRAKER including
a single ‘gene’ (W103_g1) that consists of several features.
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
V1
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
V2
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
V3
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
V4
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
V5
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
V6
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
V7
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
V8
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
V9
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
gene
</td>
<td style="text-align:right;">
13924
</td>
<td style="text-align:right;">
18566
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
W103_g1
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
transcript
</td>
<td style="text-align:right;">
13924
</td>
<td style="text-align:right;">
18566
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
W103_g1.t1
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
stop_codon
</td>
<td style="text-align:right;">
13924
</td>
<td style="text-align:right;">
13926
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
CDS
</td>
<td style="text-align:right;">
13924
</td>
<td style="text-align:right;">
13960
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
exon
</td>
<td style="text-align:right;">
13924
</td>
<td style="text-align:right;">
13960
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
intron
</td>
<td style="text-align:right;">
13961
</td>
<td style="text-align:right;">
14096
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
CDS
</td>
<td style="text-align:right;">
14097
</td>
<td style="text-align:right;">
14168
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
exon
</td>
<td style="text-align:right;">
14097
</td>
<td style="text-align:right;">
14168
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
intron
</td>
<td style="text-align:right;">
14169
</td>
<td style="text-align:right;">
14974
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
CDS
</td>
<td style="text-align:right;">
14975
</td>
<td style="text-align:right;">
15055
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
exon
</td>
<td style="text-align:right;">
14975
</td>
<td style="text-align:right;">
15055
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
intron
</td>
<td style="text-align:right;">
15056
</td>
<td style="text-align:right;">
15272
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
CDS
</td>
<td style="text-align:right;">
15273
</td>
<td style="text-align:right;">
15323
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
exon
</td>
<td style="text-align:right;">
15273
</td>
<td style="text-align:right;">
15323
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
intron
</td>
<td style="text-align:right;">
15324
</td>
<td style="text-align:right;">
15486
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
CDS
</td>
<td style="text-align:right;">
15487
</td>
<td style="text-align:right;">
15581
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
exon
</td>
<td style="text-align:right;">
15487
</td>
<td style="text-align:right;">
15581
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
intron
</td>
<td style="text-align:right;">
15582
</td>
<td style="text-align:right;">
16447
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
CDS
</td>
<td style="text-align:right;">
16448
</td>
<td style="text-align:right;">
16528
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
exon
</td>
<td style="text-align:right;">
16448
</td>
<td style="text-align:right;">
16528
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
intron
</td>
<td style="text-align:right;">
16529
</td>
<td style="text-align:right;">
16623
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
CDS
</td>
<td style="text-align:right;">
16624
</td>
<td style="text-align:right;">
16708
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
exon
</td>
<td style="text-align:right;">
16624
</td>
<td style="text-align:right;">
16708
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
intron
</td>
<td style="text-align:right;">
16709
</td>
<td style="text-align:right;">
16943
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
CDS
</td>
<td style="text-align:right;">
16944
</td>
<td style="text-align:right;">
17023
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
exon
</td>
<td style="text-align:right;">
16944
</td>
<td style="text-align:right;">
17023
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
intron
</td>
<td style="text-align:right;">
17024
</td>
<td style="text-align:right;">
17138
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
CDS
</td>
<td style="text-align:right;">
17139
</td>
<td style="text-align:right;">
17253
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
exon
</td>
<td style="text-align:right;">
17139
</td>
<td style="text-align:right;">
17253
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
intron
</td>
<td style="text-align:right;">
17254
</td>
<td style="text-align:right;">
17849
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
CDS
</td>
<td style="text-align:right;">
17850
</td>
<td style="text-align:right;">
17920
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
exon
</td>
<td style="text-align:right;">
17850
</td>
<td style="text-align:right;">
17920
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
intron
</td>
<td style="text-align:right;">
17921
</td>
<td style="text-align:right;">
18012
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
CDS
</td>
<td style="text-align:right;">
18013
</td>
<td style="text-align:right;">
18100
</td>
<td style="text-align:left;">
0.97
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
exon
</td>
<td style="text-align:right;">
18013
</td>
<td style="text-align:right;">
18100
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
intron
</td>
<td style="text-align:right;">
18101
</td>
<td style="text-align:right;">
18323
</td>
<td style="text-align:left;">
0.97
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
CDS
</td>
<td style="text-align:right;">
18324
</td>
<td style="text-align:right;">
18396
</td>
<td style="text-align:left;">
0.54
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
exon
</td>
<td style="text-align:right;">
18324
</td>
<td style="text-align:right;">
18396
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
intron
</td>
<td style="text-align:right;">
18397
</td>
<td style="text-align:right;">
18559
</td>
<td style="text-align:left;">
0.45
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
CDS
</td>
<td style="text-align:right;">
18560
</td>
<td style="text-align:right;">
18566
</td>
<td style="text-align:left;">
0.68
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
exon
</td>
<td style="text-align:right;">
18560
</td>
<td style="text-align:right;">
18566
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
<tr>
<td style="text-align:left;">
000000F|arrow
</td>
<td style="text-align:left;">
AUGUSTUS
</td>
<td style="text-align:left;">
start_codon
</td>
<td style="text-align:right;">
18564
</td>
<td style="text-align:right;">
18566
</td>
<td style="text-align:left;">
.
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
transcript_id W103_g1.t1; gene_id W103_g1;
</td>
</tr>
</tbody>
</table>
</div></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
<p>The format does not include a ‘header’ but the specification does
include names for the columns. For example, column 1 is the ‘seqid’ and
in our case is the name of the sequence (contig, chromosome, etc.).
Columns 4 and 5 are the start and end and are sorted so that column 4 is
less than column 5, this is why column 7 specifies the strand. Column 3
is the ‘type’ and is important when extracting information from the
file. The contents of this column are part of a ‘controlled vocabulary’
that we can reference at the <a
href="http://www.sequenceontology.org/">Sequence Ontology</a>. This
‘gene’ consists of 1 ‘gene’ record, 1 ‘transcript’ record, as well as 1
‘start_codon’ and one ‘stop_codon’. The gene also consists of 13 ‘CDS’,
13 ‘exon’, and 12 ‘intron’ records.</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
